### ISSUE TITLE
Add Lanczos interpolation to LookupTable2D and use that in PowerSpectrum class.

### ISSUE THREAD

**Issue by rmandelb** [GalSim-developers/GalSim#751]
Once #549 is merged, it would be nice to identify other places where we should be using LookupTable2D.  The example that comes to mind is `lensing_ps.py`, where we have some kludgy use of InterpolatedImages that would make more sense as a LookupTable2D.  Are there other cases where this would be a useful change to make?

**Comment by rmjarvis** [comment]
I'm pretty sure the only other place where we would want to use this class is the PowerSpectrum interpolation (as Rachel mentioned above).  I toyed with implementing LookupTable2D there as part of the work in PR #911.  (I did reorganize the code there significantly to use InterpolatedImage rather than SBInterpolatedImage and generally smooth out the code flow.)

The problem is that it uses Laczos interpolation and has pretty strong caveats that linear and cubic are not accurate enough, adding significant spurious power to the correlation function.  So I think in order to make the switch there, we'd need to implement Lanczos interpolation in LookupTable2D, which is a rather big job.

So we can keep this issue open if people are interested in doing that work, but I've changed the name to be more descriptive of what the job actually is.
